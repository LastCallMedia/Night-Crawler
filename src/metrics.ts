// @flow

export interface Metric {
  displayName: string;
  level: number;
  value: number;
  toString(): string;
}

export class Number implements Metric {
  displayName: string;
  level: number;
  value: number;
  constructor(label: string, level: number, value: number) {
    this.displayName = label;
    this.level = level;
    this.value = value;
  }
  toString() {
    return `${this.value}`;
  }
}

export class Percent extends Number {
  toString() {
    return `${Math.round(this.value * 100)}%`;
  }
}

export class Milliseconds extends Number {
  toString() {
    return `${Math.round(this.value)}ms`;
  }
}
