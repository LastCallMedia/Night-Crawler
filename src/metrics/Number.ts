import { Metric } from '../analysis';

export default class Number implements Metric {
  displayName: string;
  level: number;
  value: number;
  constructor(label: string, level: number, value: number) {
    this.displayName = label;
    this.level = level;
    this.value = value;
  }
  toString() {
    return `${this.value}`;
  }
}
