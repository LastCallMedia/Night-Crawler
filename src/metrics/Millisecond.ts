import Number from './Number';

export default class Milliseconds extends Number {
  toString() {
    return `${Math.round(this.value)}ms`;
  }
}
