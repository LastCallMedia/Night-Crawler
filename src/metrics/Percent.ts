import Number from './Number';

export default class Percent extends Number {
  toString() {
    return `${Math.round(this.value * 100)}%`;
  }
}
