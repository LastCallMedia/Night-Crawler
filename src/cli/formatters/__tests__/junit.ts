import formatJUnit from '../junit';
import Analysis from '../../../analysis';
import { Number } from '../../../metrics';
import fs from 'fs';
import os from 'os';

describe('JUnit formatter', function() {
  it('Should output a listing of results', function() {
    var a = new Analysis('test', new Date());
    a.addResult('OK', 0, 10, 'm1');
    a.addResult('WARN', 1, 10, 'm1');
    a.addResult('ERR', 2, 10, 'm1');
    expect(formatJUnit(a)).toMatchSnapshot();
  });
  it('Should output aggregates, marked by level', function() {
    var a = new Analysis('test', new Date());
    a.addMetric('ok', new Number('OK metric', 0, 0));
    a.addMetric('warn', new Number('WARN metric', 1, 2));
    a.addMetric('err', new Number('ERR metric', 2, 5));
    expect(formatJUnit(a)).toMatchSnapshot();
  });
  it('Should output to a file', function() {
    var filename = `${os.tmpdir()}/nightcrawler-${Math.floor(
      Math.random() * 10000
    )}`;
    var a = new Analysis('test', new Date());
    formatJUnit(a, { filename });
    expect(fs.existsSync(filename)).toEqual(true);
    fs.unlinkSync(filename);
  });
});
