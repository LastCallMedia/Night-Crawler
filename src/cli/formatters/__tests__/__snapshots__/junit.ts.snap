// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JUnit formatter Should output a listing of results 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<testsuites>
  <testsuite name=\\"Results\\" tests=\\"3\\" failures=\\"1\\" errors=\\"1\\" skipped=\\"0\\">
    <testcase classname=\\"OK\\" time=\\"0.01\\"/>
    <testcase classname=\\"WARN\\" time=\\"0.01\\">
      <error message=\\"m1\\"/>
    </testcase>
    <testcase classname=\\"ERR\\" time=\\"0.01\\">
      <failure message=\\"m1\\"/>
    </testcase>
  </testsuite>
</testsuites>"
`;

exports[`JUnit formatter Should output aggregates, marked by level 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<testsuites>
  <testsuite name=\\"Aggregates\\" tests=\\"3\\" failures=\\"1\\" errors=\\"1\\" skipped=\\"0\\">
    <testcase classname=\\"ok\\" name=\\"OK metric\\">
      <system-out>
        <![CDATA[0]]>
      </system-out>
    </testcase>
    <testcase classname=\\"warn\\" name=\\"WARN metric\\">
      <error/>
      <system-out>
        <![CDATA[2]]>
      </system-out>
    </testcase>
    <testcase classname=\\"err\\" name=\\"ERR metric\\">
      <failure/>
      <system-out>
        <![CDATA[5]]>
      </system-out>
    </testcase>
  </testsuite>
</testsuites>"
`;
